internal/service/advancedclustertpf/model_to_ClusterDescription20240805.go

```diff
@@ -93,11 +95,19 @@ func newReplicationSpec20240805(ctx context.Context, input types.List, diags *di
 			Id:            conversion.NilForUnknown(item.Id, item.Id.ValueStringPointer()),
 			ZoneId:        conversion.NilForUnknown(item.ZoneId, item.ZoneId.ValueStringPointer()),
 			RegionConfigs: newCloudRegionConfig20240805(ctx, item.RegionConfigs, diags),
-			ZoneName:      item.ZoneName.ValueStringPointer(),
+			ZoneName:      conversion.StringPtr(resolveDefaultZoneName(item)),
 		}
 	}
 	return &resp
 }
+
+func resolveDefaultZoneName(item *TFReplicationSpecsModel) string {
```

Comment:
is func name accurate? it looks like it returns the zone name or default if doesn't exist

Reply:
changing to `resolveZoneNameOrUseDefault` WDYT?

Reply:
it's fine, thanks